src/services/PlateGenerationService.ts

```diff
@@ -0,0 +1,209 @@
+            PLATES_DATA: undefined
+        };
+        const platesData = this.generateCertificateData(vehicle);
+        payload.PLATES_DATA = platesData;
+
+        // Purge undefined values
+        payload = JSON.parse(JSON.stringify(payload));
+        return payload;
+    }
+
+    /**
+     * Generates plates data for a given vehicle
+     * @param vehicle - the source vehicle for plates generation
+     */
+    public generateCertificateData(vehicle: ITechRecordWrapper) {
+        const techRecord: ITechRecord = vehicle.techRecord[0];
+        const platePayload = {
+            PlateSerialNumber: techRecord.plates.plateSerialNumber,
+            DtpNumber: techRecord.brakes.dtpNumber,
+            PrimaryVrm: vehicle.primaryVrm,
+            Vin: vehicle.vin,
+            VariantNumber: techRecord.variantNumber,
+            ApprovalTypeNumber: techRecord.approvalTypeNumber,
+            Make: techRecord.make,
+            Model: techRecord.model,
+            SpeedLimiterMrk: techRecord.vehicleType === VEHICLE_TYPES.HGV ? this.getSpeedLimiterMrk(techRecord.speedLimiterMrk) : undefined,
+            FunctionCode: techRecord.functionCode,
+            RegnDate: techRecord.regnDate,
+            ManufactureYear: this.getFormattedField(techRecord.manufactureYear),
+            GrossGbWeight: this.getFormattedField(techRecord.grossGbWeight),
+            GrossEecWeight: this.getFormattedField(techRecord.grossEecWeight),
+            GrossDesignWeight: this.getFormattedField(techRecord.grossDesignWeight),
+            TrainGbWeight: this.getFormattedField(techRecord.trainGbWeight),
+            TrainEecWeight: this.getFormattedField(techRecord.trainEecWeight),
+            TrainDesignWeight: this.getFormattedField(techRecord.trainDesignWeight),
+            MaxTrainGbWeight: this.getFormattedField(techRecord.maxTrainGbWeight),
+            MaxTrainEecWeight: this.getFormattedField(techRecord.maxTrainEecWeight),
+            Axles: this.getAxles(techRecord),
+            MaxLoadOnCoupling: techRecord.vehicleType === VEHICLE_TYPES.TRL ? this.getFormattedField(techRecord.maxLoadOnCoupling) : undefined,
+            DimensionLength: this.getFormattedField(techRecord.dimensions.length),
+            DimensionWidth: this.getFormattedField(techRecord.dimensions.width),
+            FrontAxleTo5thWheelCouplingMin: techRecord.vehicleType === VEHICLE_TYPES.HGV ? this.getFormattedField(techRecord.frontAxleTo5thWheelCouplingMin) : undefined,
+            FrontAxleTo5thWheelCouplingMax: techRecord.vehicleType === VEHICLE_TYPES.HGV ? this.getFormattedField(techRecord.frontAxleTo5thWheelCouplingMax) : undefined,
+            CouplingCenterToRearTrlMax: techRecord.vehicleType === VEHICLE_TYPES.TRL ? this.getFormattedField(techRecord.couplingCenterToRearTrlMax) : undefined,
+            CouplingCenterToRearTrlMin: techRecord.vehicleType === VEHICLE_TYPES.TRL ? this.getFormattedField(techRecord.couplingCenterToRearTrlMin) : undefined,
```

Comment:
suggest to use short circuit approach e.g.
`techRecord.vehicleType === VEHICLE_TYPES.TRL && this.getFormattedField(techRecord.maxLoadOnCoupling)`

Reply:
I cannot use this approach as the result will be "false" on the negative case and I want it to be undefined so it can be removed from the payload.

Reply:
Then wrap it into a function instead. just reduce replication footprint. 
```
function mapValue(techRecord,  requiredVehicleType, valueToMap) { 
return techRecord.vehicleType===requiredVehicleType? this. getFormattedField(valueToMap):undefined;
}
```
}

Reply:
done